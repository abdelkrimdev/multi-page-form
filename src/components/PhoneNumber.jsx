import React from 'react'
import PropTypes from 'prop-types'
import { Input, Icon } from 'antd'

const PhoneNumber = ({ onChange }) => {
  return (
    <Input
      name='phoneNumber'
      placeholder='001-541-754-3010'
      prefix={<Icon type='phone' style={{ color: 'rgba(0,0,0,.25)' }} />}
      onChange={onChange}
      size='large'
    />
  )
}

PhoneNumber.propTypes = {
  onChange: PropTypes.func
}

export default PhoneNumber
