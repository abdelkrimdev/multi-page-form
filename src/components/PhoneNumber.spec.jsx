import React from 'react'
import { shallow } from 'enzyme'

import PhoneNumber from './PhoneNumber'

describe('phone number component', () => {
  it('should render without crashing.', () => {
    const wrapper = shallow(<PhoneNumber />)

    expect(wrapper).toBeDefined()
  })

  it('should render correctly.', () => {
    const wrapper = shallow(<PhoneNumber />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should set the onChange property.', () => {
    const handleChange = jest.fn()
    const wrapper = shallow(<PhoneNumber onChange={handleChange} />)

    wrapper.prop('onChange')()

    expect(handleChange).toBeCalled()
  })
})
