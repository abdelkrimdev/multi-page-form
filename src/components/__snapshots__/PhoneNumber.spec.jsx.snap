// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`phone number component should render correctly. 1`] = `
<Input
  disabled={false}
  name="phoneNumber"
  placeholder="001-541-754-3010"
  prefix={
    <Icon
      style={
        Object {
          "color": "rgba(0,0,0,.25)",
        }
      }
      type="phone"
    />
  }
  prefixCls="ant-input"
  size="large"
  type="tel"
/>
`;
